libraries/Microsoft.Bot.Builder.Dialogs/Prompts/Prompt.cs

```diff
@@ -82,13 +86,15 @@ public override async Task<DialogTurnResult> ContinueDialogAsync(DialogContext d
             var options = (PromptOptions)instance.State[PersistedOptions];
             var recognized = await OnRecognizeAsync(dc.Context, state, options, cancellationToken).ConfigureAwait(false);
 
+            // Increment attempt count
+            state[NumberOfAttemptsKey] = (int)state[NumberOfAttemptsKey] + 1;

```

Comment:
You've switched the indexing to be 1 based rather than 0 based.

Given arrays (in JavaScript and C#) are 0 based doesn't it make more sense to have this 0 based too. For example, imagine the validator had an array of responses.

Reply:
As discussed over the phone, I renamed the property to AttemptCount and left it starting at 1.